tests/test_benchmark_cli.py

```diff
@@ -0,0 +1,193 @@
+
+    # make sure we generated the desired load function
+    with open(path) as fp:
+        expect = simpleion.load(fp, single_value=False, parse_eagerly=True)
+
+    # make sure the return values are same
+    assert actual() == expect
+
+
+@parametrize(
+    *tuple(chain(
+        generate_scalars_text(SIMPLE_SCALARS_MAP_TEXT),
+    ))
+)
+def test_generate_simpleion_dump_test_code(obj):
+    actual = generate_simpleion_dump_test_code(obj, profiling=False, binary=False)
+
+    # make sure we generated the desired dumps function
+    expect = simpleion.dumps(obj, binary=False)
+
+    # make sure the return values are same
+    assert actual() == expect
+
+
+@parametrize(
+    generate_test_path('integers.ion')
+)
+def test_read_micro_benchmark_simpleion(f):
+    file_size, result_with_gc, result_with_raw_value = \
+        read_micro_benchmark_simpleion(iterations=1, warmups=0, c_ext=True, file=f, profiling=False)
+
+    assert file_size == getsize(f) / BYTES_TO_MB
+    assert result_with_gc > 0
+    assert result_with_raw_value > 0
+
+
+@parametrize(
+    generate_test_path('integers.ion')
+)
+def test_read_micro_benchmark_iterator(f):
+    file_size, result_with_gc, result_with_raw_value = \
+        read_micro_benchmark_iterator(iterations=1, warmups=0, c_ext=True, file=f, profiling=False)
+
+    assert file_size == getsize(f) / BYTES_TO_MB
+    assert result_with_gc > 0
+    assert result_with_raw_value > 0
+
+
+def execution_with_command(c):
+    return ion_python_benchmark_cli(docopt(doc, argv=c))
+
+
+def test_option_version():
+    assert execution_with_command('-v') == TOOL_VERSION
+
+
+def test_option_write(file=generate_test_path('integers.ion')):
+    # make sure it reads successfully
+    file_size, result_with_gc, write_memory_usage_peak = \
+        execution_with_command(['write', file])
+
+    assert file_size == getsize(file) / BYTES_TO_MB
+    assert result_with_gc > 0
+    assert write_memory_usage_peak > 0
+
+
+def test_option_read(file=generate_test_path('integers.ion')):
+    # make sure it reads successfully
+    file_size, result_with_gc, conversion_time, read_memory_usage_peak = \
+        execution_with_command(['read', file])
+
+    assert file_size == getsize(file) / BYTES_TO_MB
+    assert result_with_gc > 0
+    assert read_memory_usage_peak > 0
+
+
+def test_option_write_c_extension(file=generate_test_path('integers.ion')):
+    file_size, result_with_gc, write_memory_usage_peak = \
+        execution_with_command(['write', file, '--c-extension', 'true'])
+
+    file_size_2, result_with_gc_2, write_memory_usage_peak_2 = \
+        execution_with_command(['write', file, '--c-extension', 'false'])
+
+    assert file_size == getsize(file) / BYTES_TO_MB
+    # This is a straightforward way to check if C extension is enabled from the top-level python layer.
+    # For most of the case, C extension is supposed to be faster than regular python implementation
+    assert result_with_gc < result_with_gc_2
```

Comment:
This is potentially brittle. Let's not assert on relative comparisons between result sets in the unit tests.

Reply:
Removed